src/vs/workbench/services/authentication/browser/authenticationService.ts

```diff
@@ -781,10 +781,10 @@ export class AuthenticationService extends Disposable implements IAuthentication
 		}
 	}
 
-	async removeAccountSessions(id: string, accountName: string, sessions: AuthenticationSession[]): Promise<void> {
+	async removeAccountSessions(id: string, accountName: string, sessions: AuthenticationSession[], updateAccountList: () => Promise<void> | undefined = undefined): Promise<void> { // {{SQL CARBON EDIT}} - Added updateAccountList delegate to update accounts pane after session sign out
 		const authProvider = this._authenticationProviders.get(id);
 		if (authProvider) {
-			return authProvider.removeAccountSessions(accountName, sessions);
+			return authProvider.removeAccountSessions(accountName, sessions, updateAccountList); // {{SQL CARBON EDIT}} - Added updateAccountList delegate to update accounts pane after session sign out
```

Comment:
The auth service has a `onDidChangeSessions` event - could you just use that instead (have the account list hook into that event and update itself on changes) to avoid having to modify VS Code stuff?